workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/WorkflowDirectGraphBuilder.java

```diff
@@ -153,11 +152,12 @@ private Optional<String> retrieveCondition(ActivityCompletedEvent event, BaseAct
         : Optional.ofNullable(activity.getIfCondition());
   }
 
-  private void computeSignal(WorkflowDirectGraph directGraph, Event event, String eventNodeId, int activityIndex,
-      List<Activity> activities) {
+  private void computeSignal(int activityIndex, List<Activity> activities, String eventNodeId, Event event,
+      RelationalEvents onEvents, WorkflowDirectGraph directGraph) {
     WorkflowNode signalEvent = new WorkflowNode().id(eventNodeId).event(event);
     String activityId = activities.get(activityIndex).getActivity().getId();
-    if (isFormRepliedEvent(event)) {
+
+    if (isFormRepliedEvent(event) && !event.getFormReplied().getExclusive()) {
```

Comment:
can't WorkflowNode.isNotExclusiveFormReply() be used here?

Reply:
no, it was what i did at the beginning.
the reason is because here i am trying to build this workflow node, and we don't know yet which node type this node would be, you can see the signalEvent at L157, the L160 is actually to determine the event type, so later on in the process, we will use isNotExclusiveFormReploy() method.
I even tried to put the event null check inside  isNotExclusiveFormReploy() method, but it has NPE during the process.